### PR TITLE
[release-1.18] Fix issue with dual-stack iptables6 rules when using istio-cni plugin…

### PR DESCRIPTION
**Pull request by jacob-delgado**
Fixes https://github.com/istio/istio/issues/47055

Cherry-pick of https://github.com/istio/istio/pull/46630

### CODE REVIEW

**Comment by jacob-delgado** [comment]
Still doing some testing on this. I'll remove DNM/hold when it's ready.

File: cni/pkg/plugin/plugin_test.go [hunk 1]
```diff
@@ -488,6 +488,25 @@ func TestCmdAddNoPrevResult(t *testing.T) {
 	testCmdAddWithStdinData(t, confNoPrevResult)
 }
 
+func TestCmdAddEnableDualStack(t *testing.T) {
+	defer resetGlobalTestVariables()
+	testProxyEnv["ISTIO_DUAL_STACK"] = "true"
```
Comment (adams-shaun) [inline cni/pkg/plugin/plugin_test.go:493]: `ISTIO_AGENT_DUAL_STACK`

File: tools/istio-iptables/pkg/cmd/root.go [hunk 2]
```diff
@@ -44,8 +44,8 @@ var (
 	// InvalidDropByIptables is the flag to enable invalid drop iptables rule to drop the out of window packets
 	InvalidDropByIptables = env.Register("INVALID_DROP", false,
 		"If set to true, enable the invalid drop iptables rule, default false will cause iptables reset out of window packets")
-	DualStackEnv = env.RegisterBoolVar("ISTIO_AGENT_DUAL_STACK", false,
-		"Enable pilot-agent to work in dual-stack clusters").Get()
+	DualStack = env.Register("ISTIO_DUAL_STACK", false,
```
Comment (adams-shaun) [inline tools/istio-iptables/pkg/cmd/root.go:47]: `ISTIO_AGENT_DUAL_STACK`

File: tools/istio-iptables/pkg/cmd/root.go [hunk 3]
```diff
@@ -408,6 +410,11 @@ func bindFlags(cmd *cobra.Command, args []string) {
 		handleError(err)
 	}
 	viper.SetDefault(constants.HostNSEnterExec, false)
+
+	if err := viper.BindPFlag(constants.DualStack, cmd.Flags().Lookup(constants.DualStack)); err != nil {
+		handleError(err)
+	}
+	viper.SetDefault(constants.DualStack, false)
```
Comment (adams-shaun) [inline tools/istio-iptables/pkg/cmd/root.go:417]: default to `DualStack`

**Comment by adams-shaun** [comment]
New PR here: https://github.com/istio/istio/pull/47109

**Pull request closed without merging**